python/cucim/src/cucim/skimage/measure/_colocalization.py

```diff
@@ -261,7 +255,8 @@ def manders_overlap_coeff(image0, image1, mask=None):
     if image1.min() < 0:
         raise ValueError("image1 contains negative values")
 
-    return _get_manders_overlap_coeff(image0, image1)
+    denom = cp.linalg.norm(image0) * cp.linalg.norm(image1)
+    return cp.vdot(image0, image1) / denom
```

Comment:
We could go further here and do

```python
1 - cupyx.scipy.spatial.distance.cosine(image0, image1)
```

This was added in PR ( https://github.com/cupy/cupy/pull/7063 ) and would require CuPy 12.0.0+ & pylibraft from RAPIDS

Reply:
Looks like we have to ravel to 1D to use the `cosine` solution and then do a subtraction to get the equivalent result of the linalg implementation:
```py
1 - cosine(image0.ravel(), image1.ravel())
```

If the RAFT solution is faster we can provide it with fallback to the linalg one here when unavailable.

Reply:
Sounds good. Thanks Greg! 🙏